app/controllers/api/v1/likes_controller.rb

```diff
@@ -0,0 +1,69 @@
+class Api::V1::LikesController < ApplicationController
+  before_action :set_like, only: %i[ show update destroy ]
+  include Authentication
+
+  # GET /likes
+  def index
+    likes = Like.all
+
+    render json: likes
+  end
+
+  # GET /likes/1
+  def show
+    render json: like
+  end
+
+  # POST /likes
+  def create
+    if authenticate
+      post = Post.find(params[:post_id])
+      user = User.find_by(username: request.headers[:username])
+      like = Like.new(post: post, user: user)
+
+      unless Like.where(post_id: post.id, user_id: user.id).present?
```

Comment:
Add a validation that this scenario never comes up.
Creation of likes for same post, same user should not be possible.
Then this check can be skipped. The `like.save` will throw an error in this scenario